File: data/submission/265973/C/45160370BinhNN11_70.cc
#include<bits/stdc++.h>

using namespace std;

int N;
long long M[100000];
long long p = pow(10,9) + 7;
long long solve(int n)
{
	if(M[n] != -1)
		return M[n] % p;
	else{
		M[n] = 4 * solve(n - 1) - solve(n-2);
		if(M[n] >= p)
			M[n] = M[n] % p;
	}
	return M[n] % p;
}

int main()
{
	cin >> N;
	for(int i = 0; i <= N; i++)
	{
		M[i] = -1;
	}
	M[0] = 0;
	M[1] = 1;
	long long res = solve(N);
	cout << M[N] % p;
}

File: data/submission/265973/C/45160370BinhNN11_70.java
<<<<<<< Updated upstream
N = int(input())
M = []
p = pow(10, 9) + 7
	
for i in range(N + 1):
	M.append(-1)

def solve(n):
	if M[n] != -1:
		return (M[n] % p)
	else:
		M[n] = 4 * solve(n-1) - solve(n-2)
		if M[n] > p:
			M[n] = M[n] % p
	
	return M[n]

M[0] = 0
M[1] = 1
res = solve(N)
=======
N = int(input())
M = []
p = pow(10, 9) + 7
	
for i in range(N + 1):
	M.append(-1)

def solve(n):
	if M[n] != -1:
		return (M[n] % p)
	else:
		M[n] = 4 * solve(n-1) - solve(n-2)
		if M[n] > p:
			M[n] = M[n] % p
	
	return M[n]

M[0] = 0
M[1] = 1
res = solve(N)
>>>>>>> Stashed changes
print(res % p)